app/src/androidTest/java/com/github/sdp/ratemyepfl/fragment/navigation/HomeFragmentTest.kt

```diff
@@ -0,0 +1,92 @@
+package com.github.sdp.ratemyepfl.fragment.navigation
+
+import android.content.Context
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import com.github.sdp.ratemyepfl.R
+import com.github.sdp.ratemyepfl.auth.FakeConnectedUser
+import com.github.sdp.ratemyepfl.dependencyinjection.HiltUtils
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.Rule
+import org.junit.Test
+
+@HiltAndroidTest
+class HomeFragmentTest {
+
+    @get:Rule(order = 0)
+    val hiltAndroidRule = HiltAndroidRule(this)
+
+    @get:Rule(order = 1)
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun displayTheCorrectTextWhenUserIsNotLoggedIn() {
+        FakeConnectedUser.loggedIn = false
+        FakeConnectedUser.email = null
+        val welcomeMsg = ApplicationProvider.getApplicationContext<Context>().resources.getString(
+            R.string.home_page_hello_user_text,
+            HomeFragment.VISITOR_NAME
+        )
+        HiltUtils.launchFragmentInHiltContainer<HomeFragment> {}
```

Comment:
Do we really need to use this ? Can't we just navigate to the fragment and then do what we want on it ?

Reply:
I think that each fragment needs to be tested independently from its activity. Each fragment has its logic and view and thus must be tested independently. What if we decide to modify the main activity? And what if we change the logic of the fragment? Where do I need to look to find the test? It can easily become a mess.